biotrainer/utilities/fasta.py

```diff
@@ -127,3 +129,260 @@ def read_FASTA(path: str) -> List[SeqRecord]:
+                                 - "path" (str): The dataset path.
+                                 - "sequence_column" (str): Name of the sequence column.
+                                 - "target_column" (str): Name of the target column.
+                                 - "mask_column" (Optional[str]): Name of the mask column (if applicable).
+                                 - "subset" (Optional[str]): Subset name, if required.
+
+    Returns:
+        Tuple[List[str], List[Any], Optional[List[Any]], List[str]]:
+            - List[str]: Sequences extracted from the dataset.
+            - List[Any]: targets corresponding to the sequences.
+            - Optional[List[Any]]: Masks, if available; otherwise, None.
+            - List[str]: Set values (e.g., "TRAIN", "VAL", "TEST") for each sequence.
+
+    Raises:
+        ValueError: If loading the dataset fails due to missing or invalid configuration.
+        Exception: If the required splits or columns are missing, or if other errors occur during processing.
+
+    """
+    # Extract parameters from hf_map
+    path = hf_map["path"].value
+    sequence_column = hf_map["sequence_column"].value
+    target_column = hf_map["target_column"].value
+    mask_column = hf_map["mask_column"].value if "mask_column" in hf_map else None
+    subset_name = hf_map["subset"].value if "subset" in hf_map else None
+
+    logger.info(f"Loading HuggingFace dataset from path: {path}")
+
+    # Load dataset
+    try:
+        dataset = load_dataset(path, subset_name if subset_name is not None else 'default')
+    except ValueError as e:
+        error_msg = ("Loading the dataset from Hugging Face failed. "
+                     "If the dataset requires a 'subset', you can specify it using the 'subset' option in the config file.\n"
+                     f"Error: {e}")
+        raise ValueError(error_msg)
+    except Exception as e:
+        raise Exception(f"Loading the dataset from Hugging Face failed. Error: {e}")
+
+    # Collect all available splits
+    available_splits = list(dataset.keys())
+    if not available_splits:
+        raise Exception(f"No splits found in the dataset at path '{path}'.")
+
+    sequences = []
+    targets = []
+    masks = []
+    set_values = []
+
+    if len(available_splits) == 3:
+        logger.info(f"Three splits found: {available_splits}. Processing each split separately.")
+        for split_name in available_splits:
+            split_dataset = dataset[split_name]
+            if split_dataset is None:
+                logger.warning(f"Split '{split_name}' is None. Skipping.")
+                continue
+
+            # Process split
+            split_sequences, split_targets, split_masks = process_split(split_dataset, sequence_column,
+                                                                        target_column, mask_column)
+
+            # Determine SET value based on split name
+            set_name = determine_set_name(split_name)
```

Comment:
Yes, they it can be a better naming convention for them. But I prefer not to call all of them `set_name` or something like that. Check my comment below.